src/modules/flow/http-server/http-server.c

```diff
@@ -1317,50 +1313,263 @@ blob_handle_response_cb(struct sol_flow_node *node,
+
+        r = sol_flow_packet_get_json_object(packet, &blob);
+        SOL_INT_CHECK(r, < 0, r);
+        r = replace_blob(mdata, blob);
+    } else if (type == SOL_JSON_TYPE_ARRAY_START) {
+        struct sol_blob *blob;
+
+        r = sol_flow_packet_get_json_array(packet, &blob);
+        SOL_INT_CHECK(r, < 0, r);
+        r = replace_blob(mdata, blob);
+    } else if (type == SOL_JSON_TYPE_TRUE)
+        r = boolean_process_cb(mdata, packet);
+    else if (type == SOL_JSON_TYPE_STRING)
+        r = string_process_cb(mdata, packet);
+    else if (type == SOL_JSON_TYPE_NUMBER && !is_irange)
+        r = float_process_cb(mdata, packet);
+    else if (type == SOL_JSON_TYPE_NUMBER && is_irange) {
+        struct sol_irange i;
+        struct sol_drange aux;
+
+        r = sol_flow_packet_get_irange(packet, &i);
+        SOL_INT_CHECK(r, < 0, r);
+
+        aux.val = i.val;
+        aux.min = i.min;
+        aux.max = i.max;
+        aux.step = i.step;
+
+        if (!sol_drange_equal(&mdata->value.d, &aux)) {
+            r = 1;
+            memcpy(&mdata->value.d, &aux, sizeof(struct sol_drange));
+        } else
+            r = 0;
+    } else //json NULL
+        r = 0;
+
+    SOL_INT_CHECK(r, < 0, r);
+    if (json_data->type != type) {
+        json_data->type = type;
+        return 1;
+    }
+    return r;
 }
 
 static void
 json_send_packet_cb(struct http_data *mdata,
     struct sol_flow_node *node)
 {
-    struct sol_json_scanner object_scanner, array_scanner;
+    struct http_json_data *json_data = (struct http_json_data *)mdata;
 
-    sol_json_scanner_init(&object_scanner, mdata->value.blob->mem,
-        mdata->value.blob->size);
-    sol_json_scanner_init(&array_scanner, mdata->value.blob->mem,
-        mdata->value.blob->size);
-
-    if (sol_json_is_valid_type(&object_scanner, SOL_JSON_TYPE_OBJECT_START)) {
+    switch (json_data->type) {
+    case SOL_JSON_TYPE_OBJECT_START:
         sol_flow_send_json_object_packet(node,
             SOL_FLOW_NODE_TYPE_HTTP_SERVER_JSON__OUT__OBJECT,
             mdata->value.blob);
-    } else if (sol_json_is_valid_type(&array_scanner,
-        SOL_JSON_TYPE_ARRAY_START)) {
+        break;
+    case SOL_JSON_TYPE_ARRAY_START:
         sol_flow_send_json_array_packet(node,
             SOL_FLOW_NODE_TYPE_HTTP_SERVER_JSON__OUT__ARRAY,
             mdata->value.blob);
-    } else {
+        break;
+    case SOL_JSON_TYPE_TRUE:
+    case SOL_JSON_TYPE_FALSE:
+        sol_flow_send_boolean_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_SERVER_JSON__OUT__BOOLEAN, mdata->value.b);
+        break;
+    case SOL_JSON_TYPE_STRING:
+        sol_flow_send_string_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_SERVER_JSON__OUT__STRING, mdata->value.s);
+        break;
+    case SOL_JSON_TYPE_NUMBER:
+        sol_flow_send_drange_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_SERVER_JSON__OUT__FLOAT, &mdata->value.d);
+        if (mdata->value.d.val >= INT32_MIN && mdata->value.d.val <= INT32_MAX)
```

Comment:
I thought about that too, however when I look at the json nodes I saw that they don't do that.
Then I did the same thing, to keep it consistent